src/tool-core/templates/component/t.componentdefinition.java

```diff
@@ -6,8 +6,14 @@
 
     private Map<String, Class<?>> classDirectory;
 
-    public ${self.name}(IApplication app, IRunContext runContext, int populationId) {
+    private static ${self.name} singleton;
```

Comment:
Hmmm.  Not sure we want to make this hack unconditional.  I was thinking that it would be used only when necessary (i.e., for models using WebSocket ports).  Using it unconditionally means that this and all subsequent releases of the MT version of Ciera will not be capable of supporting deployments with more than one instance of each component.  
Let's discuss this and decide how to proceed.

Reply:
We decided to leave the singleton support in both component and application along with comments explaining that any features relying on that support work only with applications with no more than a single instance of each component in the deployment.